src/RazorSdk/SourceGenerators/StaticCompilationTagHelperFeature.cs

```diff
@@ -21,24 +21,28 @@ public IReadOnlyList<TagHelperDescriptor> GetDescriptors()
                 return Array.Empty<TagHelperDescriptor>();
             }
 
-            var results = new List<TagHelperDescriptor>();
 
+            var results = new List<TagHelperDescriptor>();
 
             var context = TagHelperDescriptorProviderContext.Create(results);
             context.SetCompilation(Compilation);
-            context.Items.SetTargetMetadataReference(TargetReference);
 
-            for (var i = 0; i < _providers.Length; i++)
+            if (TargetReference is not null)
```

Comment:
When do we expect it to be non-null? Can we Debug.Assert it?

Reply:
> When do we expect it to be non-null? 

Well there was the infamous scenario that we resolved in https://github.com/dotnet/sdk/pull/16110 although that was more a bug than anything. We generally don't expect this to be null though.

> Can we Debug.Assert it?

I think it makes more sense to do a warning diagnostic so it's more visible in tooling/CLI.